scipy/stats/_stats_py.py

```diff
@@ -5551,6 +5563,159 @@ def _mgc_stat(distx, disty):
+    random_state : {None, int, `numpy.random.Generator`,
+                    `numpy.random.RandomState`}, optional
+
+        If `seed` is None (or `np.random`), the `numpy.random.RandomState`
+        singleton is used.
+        If `seed` is an int, a new ``RandomState`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``Generator`` or ``RandomState`` instance then
+        that instance is used.
+
+    Returns
+    -------
+    stat : float
+        The sample Dcorr test statistic within `[-1, 1]`.
+    pvalue : float
+        The p-value obtained via permutation.
+
+    
+    Notes
+    -----
+
+    #. Let x and y be (n,p) samples of random variables X and Y. Let D^x be the n×n 
+       distance matrix of x and D^y be the n×n be the distance matrix of y. 
+       The distance covariance is
+    
+    .. math::
+
+    Dcovnb(x,y) = (1/(n^2)) \ times trace(D^{y} H D^{y} H)
+
+    #. where tr(⋅) is the trace operator and H is defined as H=I−(1/n)J where I 
+       is the identity matrix and J is a matrix of ones. The normalized version 
+       of this covariance is distance correlation 1 and is
+    
+    .. math::
+
+    Dcorrnb(x,y) = {Dcovnb(x,y)}/{\sqrt{Dcovnb(x,x) Dcovnb(y,y)}}
+
+    #. This is a biased test statistic. An unbiased alternative also exists, 
+       and is defined using the following: 
+
+    #. Consider the centering process where 1(⋅) is the indicator function:
+    
+    .. math::
+
+    Cij^x = [\(Dsb^x - (1/(n-2)) \sum_{t=1}^{n} Dit^x - (1/(n-2)) \sum_{s=1}^{n} Dsj^x 
+             + (1/(n-2)(n+2)) \sum_{s,t=1}^{n} Dst^x]1\sb{i≠j}
+    
+    #. and similarly for Cy. 
+
+    #. Then, this unbiased Dcorr is:
+    
+    .. math:: 
+    
+    Dcovn(x,y)=(1n/(n−3))tr(CxCy)
+
+    #. The normalized version of this covariance 2 is:
+
+    .. math::
+
+    Dcorrn(x,y)={Dcovn(x,y)}/{\sqrt{Dcovn(x,x) Dcovn(y,y)}}
+
+    #. The p-value returned is calculated using a permutation test using 
+       hyppo.tools.perm_test. The fast version of the test uses hyppo.tools.chi2_approx.
+
+    #. When the data is 1 dimension and the distance metric is Euclidean, 
+       and even faster version of the algorithm is run (computational complexity is O(nlogn))^3.
+    
+    References
+    ----------
+    .. [1] Gábor J. Székely, Maria L. Rizzo, and Nail K. Bakirov. 
+           Measuring and testing dependence by correlation of distances. 
+           The Annals of Statistics, 35(6):2769–2794, December 2007. 
+           doi:10.1214/009053607000000505.
+    .. [2] Gábor J. Székely and Maria L. Rizzo. Partial distance correlation 
+           with methods for dissimilarities. The Annals of Statistics, 42(6):2382–2412, 
+           December 2014. doi:10.1214/14-AOS1255.
+    .. [3] Arin Chaudhuri and Wenhao Hu. A fast algorithm for 
+           computing distance correlation. Computational Statistics & Data Analysis, 
+           135:15–24, July 2019. doi:10.1016/j.csda.2019.01.016.
+
```

Comment:
Cite nonpar manova via independence testing paper and exact equivalence of distance and kernel methods paper